src/R/Core/Impl/AST/Extensions/AstRootExtensions.cs

```diff
@@ -38,5 +39,18 @@ public static class AstRootExtensions {
             }
             return null;
         }
+
+        public static bool IsCompleteExpression(this AstRoot expressionAst) {
+            foreach (var error in expressionAst.Errors) {
+                if (error.ErrorType == ParseErrorType.CloseCurlyBraceExpected ||
+                    error.ErrorType == ParseErrorType.CloseBraceExpected ||
+                    error.ErrorType == ParseErrorType.CloseSquareBracketExpected ||
+                    error.ErrorType == ParseErrorType.FunctionBodyExpected ||
```

Comment:
It seems that this should apply to all composite statements that have a mandatory body - e.g. `if`, `while` etc. Do we do that already? If so, this enum member should probably be renamed to `BodyExpected`.